core/client/fs/src/test/java/alluxio/client/block/stream/BlockInStreamTest.java

```diff
@@ -0,0 +1,120 @@
+import alluxio.wire.FileInfo;
+import alluxio.wire.WorkerNetAddress;
+
+import io.netty.channel.Channel;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Matchers;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+
+import java.util.Collections;
+
+/**
+ * Tests the {@link BlockInStream} class's static methods.
+ */
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({FileSystemContext.class, NettyRPC.class, NettyUtils.class})
+public class BlockInStreamTest {
+  private FileSystemContext mMockContext;
+  private BlockInfo mInfo;
+  private InStreamOptions mOptions;
+
+  @Before
+  public void before() throws Exception {
+    Channel mockChannel = PowerMockito.mock(Channel.class);
+    PowerMockito.mockStatic(NettyRPC.class);
+    PowerMockito.when(
+        NettyRPC.call(Matchers.any(NettyRPCContext.class), Matchers.any(ProtoMessage.class)))
+        .thenReturn(new ProtoMessage(Protocol.LocalBlockOpenResponse.getDefaultInstance()));
+    mMockContext = PowerMockito.mock(FileSystemContext.class);
+    PowerMockito.when(mMockContext.acquireNettyChannel(Matchers.any(WorkerNetAddress.class)))
+        .thenReturn(mockChannel);
+    PowerMockito.doNothing().when(mMockContext)
+        .releaseNettyChannel(Matchers.any(WorkerNetAddress.class), Matchers.any(Channel.class));
+    mInfo = new BlockInfo().setBlockId(1);
+    mOptions = new InStreamOptions(new URIStatus(new FileInfo().setBlockIds(Collections
+        .singletonList(1L))));
+  }
+
+  @Test
+  public void createShortCircuit() throws Exception {
+    WorkerNetAddress dataSource = new WorkerNetAddress();
+    BlockInStream.BlockInStreamSource dataSourceType = BlockInStream.BlockInStreamSource.LOCAL;
+    BlockInStream stream =
+        BlockInStream.create(mMockContext, mInfo, dataSource, dataSourceType, mOptions);
+    Assert.assertTrue(stream.isShortCircuit());
+  }
+
+  @Test
+  public void createRemote() throws Exception {
+    WorkerNetAddress dataSource = new WorkerNetAddress();
+    BlockInStream.BlockInStreamSource dataSourceType = BlockInStream.BlockInStreamSource.REMOTE;
+    BlockInStream stream =
+        BlockInStream.create(mMockContext, mInfo, dataSource, dataSourceType, mOptions);
+    Assert.assertFalse(stream.isShortCircuit());
+  }
+
+  @Test
+  public void createUfs() throws Exception {
+    WorkerNetAddress dataSource = new WorkerNetAddress();
+    BlockInStream.BlockInStreamSource dataSourceType = BlockInStream.BlockInStreamSource.UFS;
+    BlockInStream stream =
+        BlockInStream.create(mMockContext, mInfo, dataSource, dataSourceType, mOptions);
+    Assert.assertFalse(stream.isShortCircuit());
+  }
+
+  @Test
+  public void createShortCircuitDisabled() throws Exception {
+    Configuration.set(PropertyKey.USER_SHORT_CIRCUIT_ENABLED, false);
+    try {
```

Comment:
can you use configuration rule here lie this?
```java
try (Closeable c = new ConfigurationRule(PropertyKey.USER_HOSTNAME, "client").toResource()) {
}
```

Reply:
Sweet, thanks for the pointer